### DISCUSSION TITLE
[Beginner] Elements & Chrome Extension Page

### DISCUSSION THREAD

**Discussion by SirFunneh** [seleniumbase/SeleniumBase#2529] [Q&A] [answered]
Hello, trying out SeleniumBase & web scraping in general for the first time. I'm using uc mode & have 2 questions about SeleniumBase

1. Is there a way where I can print out all the elements on a page? 
I seem to be getting this error: seleniumbase.common.exceptions.NoSuchElementException: Message: Element {title} was not present after 7 seconds!

2. Is the element selectors working the same way in a chrome extension page compared to a regular chrome page? Like momentum for example? If possible could I get an example of how I can maybe type or click elements on a chrome extension page with SeleniumBase on uc mode. 

Thank you!

**Comment by mdmintz** [comment, marked as answer]
### Answer to question 1:

`elements = sb.find_elements("*")`

For example, here's a script that prints out the tag name of all the elements on a page:

```python
from seleniumbase import SB

with SB(uc=True) as sb:
    sb.open("https://seleniumbase.io/demo_page")
    elements = sb.find_elements("*")
    print(len(elements))
    for element in elements:
        print(element.tag_name)
```

### Answer to question 2:

The chrome://extensions/ page has a lot of Shadow DOM. You'll need to pierce through that to perform actions. Here are some examples that use Shadow DOM: (These use the `BaseCase` format. See [Syntax Formats](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md).)
* [examples/shadow_root_test.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/shadow_root_test.py)
* [examples/test_shadow_dom.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_shadow_dom.py)

**Reply by SirFunneh** [reply]
Hi @mdmintz ! Thanks for the answers, they have been really helpful! 

Could I ask how can I check whether uc mode is working apart from the nowsecure.nl/#relax website? And does piercing through shadow DOM elements detect uc mode? Thank you!

**Reply by mdmintz** [reply]
https://top.gg/ is another popular site with a CAPTCHA.

There are other examples too, such as:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_pixelscan.py

As for whether or not piercing through shadow DOM will make Selenium detected... that probably depends on the website.
